habitat_sim/utils/datasets_download.py

```diff
@@ -0,0 +1,252 @@
+        return
+
+    link_path = os.path.join(data_path, data_sources[uid]["link"])
+    version_tag = data_sources[uid]["version"]
+    version_dir = os.path.join(data_path, "versioned_data/" + uid + "_" + version_tag)
+
+    # check for current version
+    if os.path.exists(version_dir):
+        print(
+            f"Existing data source ({uid}) version ({version_tag}) is current. Data located: {version_dir}. Symblink: {link_path}"
+        )
+        replace_existing = (
+            replace if replace else prompt_yes_no("Replace versioned data?")
+        )
+
+        if replace_existing:
+            clean_data(uid)
+        else:
+            print("=======================================================")
+            print(
+                f"Not replacing data, generating symlink ({link_path}) and aborting download."
+            )
+            print("=======================================================")
+            # create a symlink to the versioned data
+            if os.path.exists(link_path):
+                os.unlink(link_path)
+            os.symlink(src=version_dir, dst=link_path, target_is_directory=True)
+            assert os.path.exists(link_path), "Failed, no symlink generated."
+            return
+
+    # download new version
+    download_pre_args = data_sources[uid].get("download_pre_args", "")
+    download_post_args = data_sources[uid].get("download_post_args", "")
+
+    download_command = (
+        "wget --continue "
+        + download_pre_args
+        + data_sources[uid]["source"]
+        + " -P "
+        + data_path
+        + download_post_args
+    )
+    # print(download_command)
+    os.system(download_command)
+    assert os.path.exists(
+        os.path.join(data_path, data_sources[uid]["package_name"])
+    ), "Download failed, no package found."
+
+    # unpack
+    package_name = data_sources[uid]["package_name"]
+    if package_name.endswith(".zip"):
+        with zipfile.ZipFile(data_path + package_name, "r") as zip_ref:
+            zip_ref.extractall(version_dir)
+    else:
+        # TODO: support more compression types as necessary
+        print(f"Data unpack failed for {uid}. Unsupported filetype: {package_name}")
+        return
+
+    assert os.path.exists(version_dir), "Unpacking failed, no version directory."
+
+    # create a symlink to the new versioned data
+    os.symlink(src=version_dir, dst=link_path, target_is_directory=True)
+
+    assert os.path.exists(link_path), "Unpacking failed, no symlink generated."
+
+    # clean-up
+    os.remove(data_path + package_name)
+
+    print("=======================================================")
+    print(f"Dataset ({uid}) successfully downloaded.")
+    print(f"Source: {version_dir}.")
+    print(f"Symlink: {link_path}")
+    print("=======================================================")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
```

Comment:
Nit: would be cool to have an option to print the list of available data sources.